spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/payload/JacksonResponseFieldSnippet.java

```diff
@@ -76,7 +74,13 @@ protected Type getType(final HandlerMethod method) {
         } else if ("void".equals(returnType.getName())) {
             return null;
         } else if (REACTOR_MONO_CLASS.equals(returnType.getCanonicalName())) {
-            return firstGenericType(method.getReturnType());
+            Type type = firstGenericType(method.getReturnType());
+            if (type instanceof ParameterizedType) {
+                // can be Mono<ResponseEntity<FooBar>>
+                return ((ParameterizedType) type).getActualTypeArguments()[0];
```

Comment:
Is it guaranteed that the list contains at least one element or do we need a check here?

Reply:
I very much think so, otherwise it would not be that type.